openpype/hosts/houdini/plugins/publish/collect_cache_farm.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pyblish.api
+import hou
+from openpype.hosts.houdini.api import lib, colorspace
+from openpype.hosts.houdini.api.lib import get_color_management_preferences
+
+
+class CollectDataforCache(pyblish.api.InstancePlugin):
+    """Collect data for caching to Deadline."""
+
+    order = pyblish.api.CollectorOrder + 0.04
+    families = ["ass", "pointcache",
+                "mantraifd", "redshiftproxy"]
+    hosts = ["houdini"]
+    targets = ["local", "remote"]
+    label = "Collect Data for Cache"
+    chunkSize = 999999
+
+    def process(self, instance):
+        if instance.data.get("farm", True):
+            for family in instance.data["families"]:
+                instance.data["family"] = family
+                if family == "bgeo" or "abc":
+                    instance.data["family"] = "pointcache"
+                instance.data["subset"] = instance.name
+            # instance.data["expected_families"]
+            # only created for farm caching
+            instance.data["expected_families"] = list()
+
+            instance.data["expected_families"].append(
+                instance.data["family"])
+            instance.data["families"].append("publish.hou")
+            data = dict()
+
+            data["plugin"] = "Houdini"
+            data["publish"] = True
+            ropnode = hou.node(instance.data["instance_node"])
+            instance.data.update(data)
+            output_parm = lib.get_output_parameter(ropnode)
+            # TODO: make sure the expectedfiles include all the data
+            expected_filepath = output_parm.eval()
+            if "files" not in instance.data:
+                instance.data["files"] = list()
+            if "expectedFiles" not in instance.data:
+                instance.data["expectedFiles"] = list()
+            cache_files = dict()
+            if instance.data.get("frames"):
+                files = self.get_files(
+                    instance, expected_filepath)
+                # list of files
+                instance.data["files"].extend(files)
+            else:
+                instance.data["files"].append(output_parm.eval())
```

Comment:
Note that the published representation will always then end up be integrated as a sequence since it's still a list. I believe if it's a single file than `instance.data["files"] = output_parm.eval()` the `files` data should be a single file. I'm not entirely sure but I've seen that so much in the code base that I think that's the case - it's one of the biggest annoying things haha.